src/frontend/komponenter/PersonInformasjon/useRegistrerDødsfallDatoSkjema.ts

```diff
@@ -1,110 +1,70 @@
+import { byggSuksessRessurs } from '@navikt/familie-typer';
 
+import { useRegistrerDødsfallDato } from '../../hooks/useRegistrerDødsfallDato';
 import { useBehandlingContext } from '../../sider/Fagsak/Behandling/context/BehandlingContext';
-import type { IBehandling } from '../../typer/behandling';
-import type { IManuellDødsfall } from '../../typer/dødsfall';
 import type { IGrunnlagPerson } from '../../typer/person';
-import { dateTilIsoDatoString, validerGyldigDato } from '../../utils/dato';
-import { isEmpty } from '../../utils/eøsValidators';
+import { dateTilIsoDatoString } from '../../utils/dato';
 
-interface IProps {
+interface Props {
     lukkModal: () => void;
     person: IGrunnlagPerson;
 }
 
-const erBegrunnelseFyltUt = (felt: FeltState<string>): FeltState<string> =>
-    !isEmpty(felt.verdi) ? ok(felt) : feil(felt, 'Begrunnelse for manuell registrering av dødsfall er påkrevd.');
+export enum RegistrerDødsfallDatoFelt {
+    DØDSFALL_DATO = 'dødsfallDato',
+    BEGRUNNELSE = 'begrunnelse',
+}
+
+export interface RegistrerDødsfallDatoFormValues {
+    [RegistrerDødsfallDatoFelt.DØDSFALL_DATO]: Date | null;
+    [RegistrerDødsfallDatoFelt.BEGRUNNELSE]: string;
+}
+
+type TransformedRegistrerDødsfallDatoFormValues = {
+    [RegistrerDødsfallDatoFelt.DØDSFALL_DATO]: Date;
+    [RegistrerDødsfallDatoFelt.BEGRUNNELSE]: string;
+};
 
-export const useRegistrerDødsfallDatoSkjema = ({ person, lukkModal }: IProps) => {
+export const useRegistrerDødsfallDatoSkjema = ({ person, lukkModal }: Props) => {
     const { behandling, settÅpenBehandling } = useBehandlingContext();
-    const [restFeil, settRestFeil] = useState<string | undefined>(undefined);
 
-    const {
-        skjema,
-        valideringErOk,
-        kanSendeSkjema,
-        settVisfeilmeldinger,
-        onSubmit,
-        nullstillSkjema,
-        settSubmitRessurs,
-        validerAlleSynligeFelter,
-    } = useSkjema<
-        {
-            dødsfallDato: Date | undefined;
-            begrunnelse: string;
-        },
-        IBehandling
-    >({
-        felter: {
-            dødsfallDato: useFelt<Date | undefined>({
-                verdi: undefined,
-                valideringsfunksjon: validerGyldigDato,
-            }),
-            begrunnelse: useFelt<string>({
-                verdi: '',
-                valideringsfunksjon: erBegrunnelseFyltUt,
-            }),
+    const { mutateAsync: registrerDødsfallDato } = useRegistrerDødsfallDato();
+
+    const form = useForm<RegistrerDødsfallDatoFormValues, unknown, TransformedRegistrerDødsfallDatoFormValues>({
+        values: {
+            [RegistrerDødsfallDatoFelt.DØDSFALL_DATO]: null,
+            [RegistrerDødsfallDatoFelt.BEGRUNNELSE]: '',
         },
-        skjemanavn: 'registrer-dødsfall-dato-skjema',
     });
 
-    const valideringsstatuser = [
-        skjema.felter.dødsfallDato.valideringsstatus,
-        skjema.felter.begrunnelse.valideringsstatus,
-    ];
+    const { setError } = form;
```

Comment:
Her kan det være lurt å legge til `useOnFormSubmitSuccessful` hooken slik at tilstanden i formet blir tilbakestilt ved en vellykket "submit"

```suggestion
    const { control, reset, setError } = form;

    useOnFormSubmitSuccessful(control, () => reset());
```